File: src/LambdaContainer.Core/LambdaContainer.Core.Bootloader.Provider2.Tests/Provider1.cs
using LambdaContainer.Core.Contracts;

namespace LambdaContainer.Core.BootTests.Provider2.Tests
{
    public class Provider1 : ILambdaContainerRegistry
    {
        public void WriteContentsTo(ILambdaContainerRegistrationsRecorder recorder)
        {
            recorder.Record<IFactoryRegistrations>(Read);
        }

        private void Read(ILambdaContainerRegistrationsBuilder<IFactoryRegistrations> builder)
        {
            var fullName = GetType().FullName;
            builder.Build().RegisterByName(container => fullName, fullName);
        }
    }
}

File: src/LambdaContainer.Core/LambdaContainer.Core.Bootloader.Provider1.Tests/Provider1.fs
namespace LambdaContainer.Core.BootTests.Provider1.Tests
open LambdaContainer.Core.Contracts

type public Provider1() =
    interface ILambdaContainerRegistry with
        member this.WriteContentsTo recorder =
            let typeName = typeof<Provider1>.FullName
            recorder.Record<IFactoryRegistrations>(fun builder -> builder.Build().RegisterByName( (fun _ -> typeName), typeName) |> ignore) 
            |> ignore
